src/realm/object-store/sync/sync_session.cpp

```diff
@@ -103,6 +103,27 @@ void SyncSession::become_active()
     }
 }
 
+void SyncSession::restart_session()
+{
+    util::CheckedLockGuard lock(m_state_mutex);
+    // Don't update the state if it is currently paused
+    if (m_state != State::Paused)
+        m_state = State::Inactive;
+
+    if (m_session) {
```

Comment:
Is there a reason to not call become_inactive/become_active here and go through all the normal steps that involves?

Reply:
I didn't want to cancel all the progress completions and unregister the session - the goal is to restart the session immediately and allow the progress waiters to keep waiting.

Reply:
I added a comment explaining why become_inactive() isn't used.